luacompiler/src/lib/irgen/mod.rs

```diff
@@ -61,11 +68,15 @@ impl<'a> LuaToBytecode<'a> {
                 }
             }
         }
-        LuaBytecode::new(self.instrs, self.const_map, self.reg_map.reg_count())
+        LuaBytecode::new(
+            self.instrs.iter().map(|i| i.to_32bit()).collect(),
+            self.const_map,
+            self.reg_map.reg_count() as u8,
```

Comment:
Is this cast safe?

Reply:
It isn't safe at the moment, but it will be as soon as the register allocation algorithm is implemented.

Reply:
Then it's vital that an assert is put here. We should never have unsafe code -- even temporarily unsafe code -- lurking around (because sometimes we'll forget to update it, and we'll be bitten by a bug 3 years hence).